packages/nimble-components/src/table/specs/table-column-width-hld.md

```diff
@@ -0,0 +1,157 @@
+# Table Column Widths
+
+## Problem Statement
+
+We need to provide users the means for changing the widths of individual columns as needed. This should be able to be accomplished both through interactive means as well as via a programmatic API.
+
+## Links To Relevant Work Items and Reference Material
+
+- [#873 Programmatically resize column width](https://github.com/ni/nimble/issues/873)
+- [#846 Interactively resize column width](https://github.com/ni/nimble/issues/846) 
+- [Table Design Doc](https://xd.adobe.com/view/5b476816-dad1-4671-b20a-efe796631c72-0e14/screen/d389dc1e-da4f-4a63-957b-f8b3cc9591b4/specs/)
+
+
+## Implementation / Design
+
+### Expected Behavior
+
+The behavior that has been prescribed for column sizing is as follows:
+
+- Columns should be able to be configured to either maintain a fixed width, or grow proportionally with the table such as when the window resizes causing the table width to increase. Tables can consist of columns that are configured as a mixture of the two modes.
+- If a user drags a divider between two columns to the right, then the column on the left will grow larger, and the column on the right will grow smaller by the same pixel amount. Sub-behaviors to this are:
+    - If a shrinking column has reached its minimum pixel size, then the next column in the direction of the sizing action will be affected up to the final column in a given direction.
+    - A sizing action to the left will ultimately stop having an effect when the left-most column reaches its minimum size.
+    - A sizing action to the right that would ultimately result in the final right column reaching its minimum size (all columns still within current table width) would then begin to push columns out of the table width resulting in a horizontal scrollbar.
```

Comment:
Is it possible to come up with a strategy that doesn't result in a horizontal scrollbar? It can be pretty annoying to end up having to scroll in both dimensions. I think an approach that prevents the drag from having an effect might actually be nicer, unless I'm missing a use case that would be made worse by this approach.

The SLE behavior today does result in horizontal scrollbars, though it also seems to try to avoid them in other situations like shrinking the browser window width, so I don't think the current behavior is necessarily desirable or consistent.

Reply:
Ultimately, I would rather not consider such a strategy, because it would require us to answer some seemingly impossible questions:
- Is there a maximum number of columns a user can show?
- What happens when the table shrinks in width below the combined pixel widths of all columns (enforced by their `minWidth`)?

Instead, I think it's better to provide APIs that allows users to configure their columns such that it is harder to get into situations where that will result in a horizontal scrollbar. Additionally, we've already recognized column sizing behaviors that require the horizontal scrollbar, namely the "push" behavior that AzDO grids, and many others do.

Reply:
@jattasNI I thought we had decided to allow for both behaviors, as there are likely use cases for both across web and desktop.

Reply:
I'll defer to the recommendation from @atmgrifter00. With the diversity of situations/apps that our table will get used in (and the unknowable amount columns an application will need to display), I think it will be futile for us to try to create a table that will never have horizontal scrollbars (happy to be wrong about that, though). 

I appreciate that our current direction from @atmgrifter00 seems to make it more difficult to end up in that situation but I think it will be impossible to 100% eliminate (again, happy to be wrong). For now, I think the best approach is to do something that is:

- Relatively straightforward for us to implement
- Easy to configure for our user
- Intuitive/Learnable to use and master by our end user

It might be helpful to see that solution in a prototype as the intricacies of these interactions and diversity of situations make it pretty challenging to visualize without interacting with something more directly.

Reply:
I defer to the other opinions on this thread. We can always revisit this if we get client feedback.